tensorflow_lite_support/cc/task/processor/image_postprocessor.cc

```diff
@@ -0,0 +1,129 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/cc/task/processor/image_postprocessor.h"
+
+namespace tflite {
+namespace task {
+namespace processor {
+
+namespace {
+
+using ::absl::StatusCode;
+using ::tflite::metadata::ModelMetadataExtractor;
+using ::tflite::support::CreateStatusWithPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+
+}  // namespace
+
+/* static */
+tflite::support::StatusOr<std::unique_ptr<ImagePostprocessor>>
+ImagePostprocessor::Create(core::TfLiteEngine* engine, const int output_index,
+                           const int input_index) {
+  ASSIGN_OR_RETURN(auto processor,
+                   Processor::Create<ImagePostprocessor>(
+                       /* num_expected_tensors = */ 1, engine, {output_index},
+                       /* requires_metadata = */ false));
+
+  RETURN_IF_ERROR(processor->Init(input_index));
+  return processor;
+}
+
+absl::Status ImagePostprocessor::Init(const int input_index) {
+  if (input_index == -1) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat("Input image tensor not set. Input index found: %d",
+                        input_index),
+        tflite::support::TfLiteSupportStatus::kInputTensorNotFoundError);
+  }
+  ASSIGN_OR_RETURN(
+      auto output_specs,
+      vision::BuildImageTensorSpecs(*engine_->interpreter(),
+                                    *engine_->metadata_extractor(), false));
+  options_ = std::make_unique<vision::NormalizationOptions>(
+      output_specs.normalization_options.value());
+  return absl::OkStatus();
+}
+
+absl::StatusOr<vision::FrameBuffer> ImagePostprocessor::Postprocess() {
+  has_uint8_output_ = GetTensor()->type == kTfLiteUInt8;
+  const int kRgbPixelBytes = 3;
+
+  vision::FrameBuffer::Dimension to_buffer_dimension = {
+      GetTensor()->dims->data[2], GetTensor()->dims->data[1]};
+  size_t output_byte_size =
+      GetBufferByteSize(to_buffer_dimension, vision::FrameBuffer::Format::kRGB);
+  std::vector<uint8> postprocessed_data(output_byte_size / sizeof(uint8), 0);
+
+  if (has_uint8_output_) {  // No denormalization required.
+    if (GetTensor()->bytes != output_byte_size) {
+      return tflite::support::CreateStatusWithPayload(
+          absl::StatusCode::kInternal,
+          "Size mismatch or unsupported padding bytes between pixel data "
+          "and output tensor.");
+    }
```

Comment:
This size check is redundant, because both GetTensor()->bytes and output_byte_size are calculated from output tensor shape. They are from the same ground truth.